framework/include/utils/MooseTypes.h

```diff
@@ -166,6 +166,8 @@ enum KernelType
 {
   KT_TIME = 0,
   KT_NONTIME = 1,
+  KT_NONEIGEN = 2,
+  KT_EIGEN  = 3,
```

Comment:
We use them in a few places: `FEProblemBase`, `ComputeResidualThread` and `ComputeJacobianThread`.

Following the existing implementation for time kernels, I have to handle the eigenvalue kernels like this way. Any better way are welcome.